test-project/Assets/EditmodeTests/Utility/InterestConstraintTests.cs

```diff
@@ -0,0 +1,115 @@
+using Improbable.Gdk.Core;
+using Improbable.Gdk.QueryBasedInterest;
+using NUnit.Framework;
+using UnityEngine;
+
+namespace Improbable.Gdk.EditmodeTests.Utility
+{
+    [TestFixture]
+    public class InterestConstraintTests
+    {
+        private const double Distance = 10;
+        private readonly Vector3 _center = Vector3.zero;
+
+        private static ComponentInterest.QueryConstraint BasicConstraint()
+        {
+            return Constraint.Component<Position.Component>();
+        }
+
+        [Test]
+        public void All_constraint_sets_OrConstraint_to_empty_list()
+        {
+            var constraint = Constraint.All(BasicConstraint());
+            Assert.True(constraint.AndConstraint.Count > 0);
+            Assert.True(constraint.OrConstraint.Count == 0);
+        }
+
+        [Test]
+        public void All_constraint_with_no_params_sets_AndOr_constraints_to_empty_list()
+        {
+            var constraint = Constraint.All();
```

Comment:
My thinking is that it is simpler to use `params cs[]` once and handle the empty array case, instead of doing `constraint c, params cs[]`

(similar with the queries case earlier)

Reply:
Its not clear what this as an API should do though

Reply:
should be clearer now

also removed AddQuery<T> so all queries must be added via AddQueries<T>